mlir/lib/Catalyst/Transforms/InlineNestedModules.cpp

```diff
@@ -183,14 +185,17 @@ struct AnnotateWithFullyQualifiedName : public OpInterfaceRewritePattern<SymbolO
 
 struct RenameFunctionsPattern : public RewritePattern {
     /// This overload constructs a pattern that matches any operation type.
-    RenameFunctionsPattern(MLIRContext *context, SmallVector<Operation *> *symbolTables)
-        : RewritePattern(MatchAnyOpTypeTag(), 1, context), _symbolTables(symbolTables)
+    RenameFunctionsPattern(MLIRContext *context, SmallVector<Operation *> *symbolTables,
+                           llvm::SmallSet<StringRef, 8> *externalFuncDeclNames)
+        : RewritePattern(MatchAnyOpTypeTag(), 1, context), _symbolTables(symbolTables),
+          _externalFuncDeclNames(externalFuncDeclNames)
     {
     }
 
     LogicalResult matchAndRewrite(Operation *op, PatternRewriter &rewriter) const override;
 
     SmallVector<Operation *> *_symbolTables;
+    llvm::SmallSet<StringRef, 8> *_externalFuncDeclNames;
```

Comment:
This is just me being paranoid, but since we've had so many incidents with object lifetimes I want to confirm with you that the string storage lives for as long as this set does?